echion/threads.h

```diff
@@ -73,12 +94,10 @@ class ThreadInfo
         : thread_id(thread_id), native_id(native_id), name(name)
     {
 #if defined PL_LINUX
-        // Try to check that the thread_id is a valid pointer to a pthread
-        // structure. Calling pthread_getcpuclockid on an invalid memory address
-        // will cause a segmentation fault.
-        char buffer[32] = "";
-        if (copy_generic((void *)thread_id, buffer, sizeof(buffer)))
-            throw Error();
+        char buffer[64];
+        snprintf(buffer, sizeof(buffer), "/proc/self/task/%lu/stat", native_id);
+        stat_fd = std::make_unique<fd>(open(buffer, O_RDONLY));
```

Comment:
Does this risk exhausting FDs if we have many threads?

Reply:
It is a possibility, but I'd say it would be uncommon for a Python applications to have many threads per process. For I/O-intensive applications, I'd expect people to use `asyncio`, which is in general a single thread per process. In other scenarios, we would have multiple worker _processes_, with a limited number of threads per process (again, generally just one). Our tracer contributes threads of course, but:
1. we generally try to limit the number of Python threads we spawn (per process)
2. we're slowly moving away from Python threads anyway

Reply:
We could add a cap to the fds we use for tracked threads per process as a safeguard measure though

Reply:
We could cap the number of fds we open instead, and if we saturate that we fall back to the current behaviour. This way we still track all threads. I have a change ready with this that I'll get out later on today

Reply:
FWIW, 1024 is the lowest common file descriptor limit I'd anticipate on contemporary systems--and that's almost exclusively to support extensions that still rely on `select()` (there are a few of them IIRC).  `select()` has a kernel build-time limit on the _largest file descriptor_ it will scan for.  Anything larger will be invisible.

This comment is less about exhaustion and more about the introduction of unwanted behavior in the overall application.

Reply:
Now we have configurable maximum number of file descriptors to use. Please take a look again.